src/components/FilterSizes.js

```diff
@@ -0,0 +1,114 @@
+import React from 'react';
+import size from '../img/size.png';
+import './style/FilterSizes.css';
+
+
+
+
+class FilterSizes extends React.Component {
+    //when you hover over a div, all others become visible and text pops up
+    hover = event => {
+        const popups = document.getElementsByClassName("popup");
+        for(let i = 0; i < popups.length; i++) {
+            popups[i].style.opacity = 0.9;
+        }
+        event.target.style.opacity = 0.1;
+        switch(event.target.id) {
+            case "one":
+                document.getElementById("input").style.zIndex = "3";
+                document.getElementById("input").style.opacity = 1;
+                break;
+            case "two":
+                document.getElementById("convolve").style.zIndex = "3";
+                document.getElementById("convolve").style.opacity = 1;
+                break;
+            case "three":
+                document.getElementById("filters").style.zIndex = "3";
+                document.getElementById("filters").style.opacity = 1;
+                break;
+            case "four":
+                document.getElementById("output-separate").style.zIndex = "3";
+                document.getElementById("output-separate").style.opacity = 1;
+                break;
+            case "five":
+                document.getElementById("output-together").style.zIndex = "3";
+                document.getElementById("output-together").style.opacity = 1;
+                break;
+            default:
+                break;
+        }
+    }
```

Comment:
I think for this code block, manipulating the DOM with `document` is not the way to go - it messes with React internals and also can lead to some buggy code! I would instead have some component state that checks whether or not any component is being hovered; then, you can toggle specific CSS properties (e.g. `z-index`, `opacity`) depending on if the element is being hovered. Not sure if you've tried this already though!

Reply:
So using the state-based approach would it be appropriate to toggle the style using CSS in JS?